src/pages/ExpiringResults/index.js

```diff
@@ -0,0 +1,295 @@
+    const updatedResult = totalResultData.filter(item => !keys.includes(item.key));
+    dispatch({
+      type: 'dashboard/updateResults',
+      payload: updatedResult,
+    });
+  };
+
+  removeResultFromSeen = controller => {
+    console.log(controller);
+    const { dispatch } = this.props;
+    dispatch({
+      type: 'user/removeResultFromSeen',
+      payload: controller,
+    });
+  };
+
+  showDrowpdown = id => {
+    const dropdownElem = document.getElementById(id);
+    if (dropdownElem.style.display === 'none') {
+      dropdownElem.style.display = 'block';
+    } else {
+      dropdownElem.style.display = 'none';
+    }
+  };
+
+  render() {
+    const { favoriteControllers, seenResults } = this.props;
+    const { totalResultData } = this.state;
+    const seenDataColumns = [
+      {
+        Header: 'Result',
+        accessor: 'result',
+        Cell: cell => {
+          const row = cell.row.original;
+          let isSeen = false;
+          if (seenResults !== []) {
+            seenResults.forEach(item => {
+              if (item.original.key === row.key) {
+                isSeen = true;
+              }
+            });
+          }
+          if (isSeen) {
+            return (
+              <div>
+                <Button
+                  variant="link"
+                  isInline
+                  style={{ marginBottom: '8px' }}
+                  onClick={() => this.retrieveResults(cell.row)}
+                >
+                  {cell.value}
+                </Button>
+                <br />
+                <Text component={TextVariants.p} className={styles.subText}>
+                  <span className={styles.label}>{row.controller}</span>
+                </Text>
+              </div>
+            );
+          }
+          return (
+            <div>
+              <Button
+                variant="link"
+                isInline
+                style={{ marginBottom: '8px' }}
+                onClick={() => this.retrieveResults(cell.row)}
+              >
+                <b>
+                  <i>{cell.value}</i>
+                </b>
+              </Button>
+              <br />
+              <Text component={TextVariants.p} className={styles.subText}>
+                <span className={styles.label}>{row.controller}</span>
+              </Text>
+            </div>
+          );
+        },
+      },
+      {
+        Header: 'End Time',
+        accessor: 'end',
+        Cell: cell => (
+          <Tooltip content={formatDate('utc', cell.value)}>
+            <span>{formatDate('with time', cell.value)}</span>
+          </Tooltip>
+        ),
+      },
+      {
+        Header: 'Scheduled for deletion on',
+        accessor: 'deletion',
+        Cell: cell => {
+          const remainingDays = getDiffDays(cell.value);
+          if (remainingDays > 15) {
+            return (
+              <div>
+                <Tooltip content={formatDate('without time', cell.value)}>
```

Comment:
We can move the conditional further into the markdown to avoid repeating code. For example: 

```
          return (
              <div>{remainingDays > 15 ? (
                <Tooltip content={formatDate('without time', cell.value)}>
<span>{getDiffDate(cell.value)}</span>
                </Tooltip>) : (<span>{cell.value}</span>)}
                <Progress
                  min={0}
                  max={expirationLimit.val}
                  value={expirationLimit.val - remainingDays}
                  size={ProgressSize.sm}
                  measureLocation={ProgressMeasureLocation.none}
                />
              </div>
            );
          }
```

Reply:
done !